platform/linux-generic/odp_pool.c

```diff
@@ -641,9 +642,9 @@ static odp_pool_t pool_create(const char *name, odp_pool_param_t *params,
 		block_size = ROUNDUP_CACHE_LINE(hdr_size + align_pad + seg_len);
 	}
 
-	/* Allocate extra memory for skipping packet buffers which cross huge
+	/* Allocate extra memory for skipping buffers which cross huge
 	 * page boundaries. */
-	if (params->type == ODP_POOL_PACKET) {
+	if (params->skip_cross_page) {
```

Comment:
At a quick glance it seems we could make this the default operation for all pool types, so no API changes would be required.

Reply:
only in the situation:
   buffer is shared to kernel space .  kernel convert phy-addr to virt-addr to access memory, it ask buffer don't cross page .
Generally , need not to skip cross page to save memory .

Reply:
This could be the default operation with huge pages for all pool types. E.g. with 2MB huge page size and 2kB buffer size the memory loss would 1/1000, which is not very significant.

Reply:
ok ,let's  make it default .